docs/plugins.md

```diff
@@ -85,64 +85,224 @@ the output of Protractor by returning a results object.
+ *     rejected, a failed assertion is added to the test results.
  */
-exports.postResults = function(config) {};
+exports.postResults = function() {};
 
 /**
  * Called after each test block (in Jasmine, this means an `it` block)
  * completes.
  *
- * @param {Object} config The plugin configuration object.
  * @param {boolean} passed True if the test passed.
  * @param {Object} testInfo information about the test which just ran.
  *
- * @return Object If an object is returned, it is merged with the Protractor
- *     result object. May return a promise.
+ * @this {Object} bound to module.exports
+ *
+ * @throws {*} If this function throws an error, a failed assertion is added to
+ *     the test results.
+ *
+ * @return {q.Promise=} Can return a promise, in which case protractor will wait
+ *     for the promise to resolve before outputting test results.  Protractor
+ *     will *not* wait before executing the next test, however.  If the promise
+ *     is rejected, a failed assertion is added to the test results.
+ */
+exports.postTest = function(passed, testInfo) {};
+
+/**
+ * This is called inside browser.get() and allows the plugin to modify any URLs
+ * the tests direct the browser to.
+ *
+ * @param {string} url The unmodified URL which the test called browser.get()
+ *     with
+ *
+ * @this {Object} bound to module.exports
+ *
+ * @throws {*} If this function throws an error, a failed assertion is added to
+ *     the test results and the original URL is used
+ *
+ * @return {string|q.Promise<string>} The modified URL or a promise which
+ *     resolves to the modified URL.  If a rejected promise is returned, a
+ *     failed assertion is added to the test results.
+ */
+exports.rewriteURL = function() {};
+
+/**
+ * This is called inside browser.get() directly after the page loads, and before
+ * angular has had time to bootstrap/synchronize. 
+ *
+ * @this {Object} bound to module.exports
+ *
+ * @throws {*} If this function throws an error, a failed assertion is added to
+ *     the test results.
+ *
+ * @return {q.Promise=} Can return a promise, in which case protractor will wait
+ *     for the promise to resolve before continuing.  If the promise is
+ *     rejected, a failed assertion is added to the test results.
+ */
+exports.onPageLoad = function() {};
+
+/**
+ * This is called inside browser.get() directly after angular is done
+ * bootstrapping/synchronizing.  If browser.ignoreSynchronization is true, this
+ * will not be called.
+ *
+ * @this {Object} bound to module.exports
+ *
+ * @throws {*} If this function throws an error, a failed assertion is added to
+ *     the test results.
+ *
+ * @return {q.Promise=} Can return a promise, in which case protractor will wait
+ *     for the promise to resolve before continuing.  If the promise is
+ *     rejected, a failed assertion is added to the test results.
  */
-exports.postTest = function(config, passed, testInfo) {};
+exports.onPageSync = function() {};
```

Comment:
Let's call this `onPageStable` to try to avoid overloading the `sync` term more. That also corresponds with the `whenStable` call we make to Angular Testability.